### ISSUE TITLE
Start-Process doesn't pass arguments with embedded whitespace and double quotes correctly

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#5576]
<sup>Updated later to include the problem with embedded double quotes.</sup>

Steps to reproduce
------------------

Embedded whitespace:

```powershell
'"Hi!"' > './t 1.ps1'; Start-Process -Wait -NoNewWindow pwsh -ArgumentList '-noprofile', '-file', './t 1.ps1'
```

Embedded double quotes:

```powershell
Start-Process -Wait -NoNewWindow pwsh -ArgumentList '-noprofile', '-command', '"Hi!"'
```

Expected behavior
-----------------

In both cases:

```none
Hi!
```

That is, script file `./t 1.ps1` should execute, and double-quoted string literal `"Hi!"` should print.

Actual behavior
---------------

Embedded whitespace:

Invocation fails, because the `./t 1.ps1` is passed as _two_ arguments:

```none
The argument './t' is not recognized as the name of a script file. Check the spelling of the name, or if a path was included, verify that the path is
correct and try again.
```

The only way to make this currently work is to _embed_ (potentially escaped) _double quotes_: `'"./t 1.ps1"'` or ``"`"./t 1.ps1`""``; e.g.:

```powershell
Start-Process -Wait -NoNewWindow pwsh -ArgumentList '-noprofile', '-file', "`"./t 1.ps1`""
```

_Update_: Overall, the **best workaround** is to pass a _single_ string containing _all_ arguments to `-ArgumentList` and use embedded quoting and escaping as necessary:

```powershell
Start-Process -Wait -NoNewWindow pwsh -ArgumentList '-noprofile -file "./t 1.ps1"'
```

Embedded double quotes:

The embedded double quotes are unexpectedly _removed_.

The only way to make this currently work is to `\`-escape the embedded double quotes: 

```powershell
Start-Process -Wait -NoNewWindow pwsh -ArgumentList '-noprofile', '-command', '\"Hi!\"'
```

_Update_: Again, the best workaround is to use a _single_ string:

```powershell
Start-Process -Wait -NoNewWindow pwsh -ArgumentList '-noprofile -command \"Hi!\"'
```

Environment data
----------------

```powershell
PowerShell Core v6.0.0-rc (v6.0.0-rc) on Microsoft Windows 7 Enterprise  (64-bit; v6.1.7601)
Windows PowerShell v5.1.14409.1012 on Microsoft Windows 7 Enterprise  (64-bit; v6.1.7601)
```

**Comment by ZSkycat** [comment]
I think it just executes `{filename} {ArgumentList.join(' ')}`

try this code
```powershell
Start-Process -Wait -NoNewWindow pwsh.exe -ArgumentList '-noprofile', '-file', '"./t 1.ps1"'
Start-Process -Wait -NoNewWindow pwsh.exe -ArgumentList '-noprofile', '-file', '"', './t 1.ps1', '"'
```

**Comment by bergmeister** [comment]
The code where it deconstructs `-Argumentlist` to the Argument string for `ProcessStartInfo.Arguments` [here](https://github.com/PowerShell/PowerShell/blob/4bc52d2358222084738157a08907fac32d13bd3a/src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs#L1863) and it simply joins the arguments by separating it with a space character. Therefore it sees `./t` and `1.ps1` as 2 separate arguments. As pointed out by @ZSkycat one has to correctly quote the file. Therefore I would not qualify this as a bug but rather as by design. WDYT @SteveL-MSFT

**Comment by iSazonov** [comment]
I see the [feature is documented](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.arguments?view=netcore-2.0):
>Arguments are parsed and interpreted by the target application, so must align with the expectations of that application. For.NET applications as demonstrated in the Examples below, spaces are interpreted as a separator between multiple arguments. A single argument that includes spaces must be surrounded by quotation marks, but those quotation marks are not carried through to the target application. In include quotation marks in the final parsed argument, triple-escape each mark.

I don't understand how we can fix this for all platforms and for an indefinite number of applications. If there is no standard to which we should follow I would rather expect a common way of passing arguments so that they reach the goal application in exactly the user specified form.

**Comment by bergmeister** [comment]
Well, I guess we can only improve it (see my PR 5703) or maybe the better solution is to add an `-Arguments` parameter that is just a string to reduce complexity and leave the `-ArgumentList` parameter there only for legacy reasons.

**Comment by mklement0** [comment]
@bergmeister: Thanks for taking this on in your PR, but there are additional edge cases to consider, such as tokens ending in `\` - see https://github.com/PowerShell/PowerShell/issues/1995#issuecomment-330675421

Ultimately, the single string passed to `ProcessStartInfo.Arguments` should be constructed the same way that a direct call to an external program is handled, namely based on inverse of the [MSVC++ command-line parsing rules](https://docs.microsoft.com/en-us/cpp/cpp/parsing-cpp-command-line-arguments), as discussed in @TSlivede's [RFC proposal](https://github.com/PowerShell/PowerShell-RFC/pull/90).

Longer-term, once https://github.com/dotnet/corefx/issues/23592 is implemented in CoreFx, PowerShell will simply be able to pass the array _through_. The linked issue also points to a utility method in the CoreFx code where the kind of array-to-command-line transformation that would be needed here is used internally.

And, yes, this will break things, inevitably and in multiple scenarios, but I think if PowerShell wants to be taken seriously as a multi-platform shell, there's no way around that.

While we may consider adding a new `-Arguments` parameter for someone who wants to pass a _single-string_, _pre-escaped_ command line (directly assignable to `ProcessStartInfo.Arguments` - which, notably, still needs to be split back into an array _before_ creating a process on _Unix_), it is `-ArgumentList` that must be fixed - which breaks things.

**Comment by iSazonov** [comment]
@mklement0  As far as I understand the shortest way to get this is to implement https://github.com/dotnet/corefx/issues/23592, isn't it?

**Comment by mklement0** [comment]
@iSazonov: Yes, with said proposal implemented, PowerShell could just use the new `ProcessStartInfo.ArgumentList` property directly and, on Windows, let CoreFX translate that into a single command line string with appropriate quoting.

**Comment by bergmeister** [comment]
Ok. Shall I close the PR then?

**Comment by iSazonov** [comment]
I think it is not critical and not a secure hole so we can wait and it is better to direct our efforts to other areas or contribute directly in CoreFX now.

**Comment by Artoria2e5** [comment]
On a related note, it seems to truncate at the `>` character too:
```
PS> Start-Process -Wait -NoNewWindow node -ArgumentList '-e', 'process.argv.slice(1).forEach((x) => console.log(x))'
[eval]:1
process.argv.slice(1).forEach((x)
                                ^
SyntaxError: missing ) after argument list
```
(Windows PS 7.0.0-preview3)

**Comment by mklement0** [comment]
@Artoria2e5: The problem isn't `>`, the problem is that `process.argv.slice(1).forEach((x) => console.log(x))` isn't passed through as a _single argument_, it is broken into _3_ arguments by _whitespace_.

That is, your `node` command is effectively executed as follows, which explains the symptom:

```
node -e 'process.argv.slice(1).forEach((x)' '=>' 'console.log(x))'
```

**Comment by Artoria2e5** [comment]
ho boy...

**Comment by felixfbecker** [comment]
This is very annoying when wrapping Linux command line tools. Could this be considered for PowerShell 7? It seems like the fix would be simple (just use `ProcessStartInfo.ArgumentList`). Using `System.Diagnostics.Process` in PowerShell directly works as expected, but is incredibly cumbersome.

**Comment by rjmholt** [comment]
> It seems like the fix would be simple (just use ProcessStartInfo.ArgumentList

Having used `ProcessStartInfo.Arguments` recently (had to support netstandard2.0), it's a very hard API to use.

I'd second the recommendation to switch to ArgumentList if we can.

**Comment by SteveL-MSFT** [comment]
I have a PR to fix this, just need to add tests.

In the second example:

```powershell
Start-Process -Wait -NoNewWindow pwsh -ArgumentList '-noprofile', '-command', '"Hi!"'
```

This becomes: `pwsh -noprofile -command "Hi!"` so it should only print out `Hi!` without the quotes.  The quotes are needed to tell PowerShell this is a string and not a command.  After this fix, you still need to have nested double quotes if you want quotes:

```powershell
Start-Process -Wait -NoNewWindow pwsh -ArgumentList '-noprofile', '-command', '"""Hi!"""'
```

**Comment by mklement0** [comment]
Glad to see that this is getting tackled.

> This becomes: `pwsh -noprofile -command "Hi!"` so it should only print out Hi! without the quotes.

As a _shell command_, `pwsh -noprofile -command "Hi!"`,  _fails_, because the the closing `"` are removed during argument parsing, causing PowerShell to attempt to execute `Hi!`.

What `Start-Process -Wait -NoNewWindow pwsh -ArgumentList '-noprofile', '-command', '"Hi!"'` _should_ translate to is:

On Windows:

```
pwsh -noprofile -command "\"Hi!\""
```

On Unix, the array of verbatim tokens needs to be:

`pwsh`, `-noprofile`, `-command`, `"Hi!"`

The collection-based [`ProcessStartInfo.ArgumentList`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.argumentlist) - not the single-string `ProcessStartInfo.Arguments` - should do all that for us.

**Comment by SteveL-MSFT** [comment]
Unfortunately, some bad news.  Use of `ArgumentList` makes this a breaking change.  Previously, you could do something like:

```powershell
start-process ping -argumentlist "-n 2 localhost"
```

But this becomes:

```
ping "-n 2 localhost"
```

This means I can't rely on `ArgumentList` and will have to quote things correctly in the code as a string...

On the topic of:

```powershell
Start-Process -Wait -NoNewWindow pwsh -ArgumentList '-noprofile', '-command', '"Hi!"'
```

Again, this becomes:

```
pwsh -noprofile -command "Hi!"
```

We could double escape the quotes to get the desired behavior, but then you can't get the result above.

**Comment by vexx32** [comment]
From a pure UX perspective, I think it would be much better to have `-ArgumentList` be strictly an actual **list** of arguments (so, it would map directly to `ArgumentList` in the code), and then introduce an additional parameter that mimics legacy behaviour.

Yes, it would break something, but given the difficulty of using the current behaviour correctly... I think we're rather better off breaking it here. 😕

**Comment by bergmeister** [comment]
Although I wish the parameter behaviour was better, I agree with Steve, that we cannot make a breaking change in such an important cmdlet because it is the foundation of a lot of code and use cases are quite often very generic. A classic example is the software of CI agents that needs to spawn processes, this software usually delegates some of its work from .net/java/C++ to something simpler like node/python/powershell for doing that. Because this piece of software is implicitly used by other users, it is not known how users will call into it and implicitly break peoples build or release pipelines.
The change will need to be in a non-breaking way or at least have some way to opt back into legacy behaviour.

**Comment by SteveL-MSFT** [comment]
It's way too likely people are doing:

```powershell
start-process ping -argumentlist "-n",2,"localhost"
start-process ping -argumentlist "-n 2","localhost"
start-process ping -argumentlist "-n 2 localhost"
```

where all 3 have the same behavior today since it's a simple concatenation with whitespace in between.  This is apparent to me looking at the test failures in my attempt to use `ProcessStartInfo.ArgumentList` that our tests do this.  Maybe `-LiteralArgumentList` is ok.

**Comment by mklement0** [comment]
Good points, @SteveL-MSFT  and @bergmeister, as much as I wish we could go with @vexx32's suggestion.

It does seem like a new parameter is the only non-breaking solution, but, to better reflect the distinction, I propose the following names

* `-IndividualArguments` 
* alias `-iArgs` 

Additionally, we could consider renaming `-ArgumentList` to `-Arguments`, making the former an alias of the latter, and highlighting the latter in the help topic (the `-Args` alias luckily already has the better name).

**Comment by felixfbecker** [comment]
It's also not just `Start-Process`, argument splatting is also affected. I recently had this problem where I was calling a utility on macOS like `foo @bar`. One would expect this to call `foo` with exactly the array `$bar` as arguments, instead it joins the array into a string, then splits the string by spaces and passes that array as arguments. This broke the tool whenever I wanted to pass a string as first parameter that contained a space.

A different parameter for `Start-Process` would not solve this usage, and it would be really annoying if we basically have to tell macOS and Linux users they cannot rely on calling commands directly and always need to use `Start-Process`.

**Comment by SteveL-MSFT** [comment]
@felixfbecker that's a different issue than this one, can you open a new issue or see if there is an existing one for that?

**Comment by vexx32** [comment]
That seems like the same problem with #1995 (or at least the same possible fix, anyway).

**Comment by mklement0** [comment]
I agree that splatting is incidental to the issue and that it comes down to #1995:

Assume a `de.exe` executable that prints the full command line with which it is invoked:

```powershell
PS> de.exe '"hi there"' 
"c:\path\to\de.exe" "hi there"   # broken: embedded " weren't preserved

# Ditto with splatting  (or direct passing of $a)
PS> $a = , '"hi there"'; de.exe @a  
"c:\path\to\de.exe" "hi there"
```

Yes, passing arguments with embedded double quotes is fundamentally broken, and always has been - because of existing workarounds, it's a problem to fix it now without breaking lots of code.

@felixfbecker, please start reading at https://github.com/PowerShell/PowerShell/issues/1995#issuecomment-552223508 for the current state of the debate.

**Comment by iSazonov** [comment]
Since #1995 is still a big question, we definitely need to resolve the issue with Start-Process in the 7.1 milestone.

I believe we can add a new `-Arguments` parameter. Internally, it will use `ProcessStartInfo.ArgumentList`, but this internal implementation is invisible to users and should not confuse them.

**Comment by iSazonov** [comment]
I think it is better to avoid unneeded complicity - best is the enemy of good.

> We also need to think about a short parameter alias that parallels -Args - what would that be for -Arguments?

With IntelliSense and tab-completion it is minor.

> While PowerShell in general needn't mirror the underlying .NET APIs, I find it problematic that the semantics in this case would be the exact opposite of the underlying .NET API's

Yes, PowerShell users do not see and don't think about  underlying .NET API's. For developers we add docs and comments too.

If we blog post and enhance PSSA to recommend the new parameter I believe user adaptation will be easy.

**Comment by mklement0** [comment]
@iSazonov 

> better to avoid unneeded complicity

What unneeded complexity? If you're referring to re-typing `-ArgumentList` to `string`, there is minor implementation complexity and no added complexity to end users - on the contrary, it makes things clearer for them.

> With IntelliSense and tab-completion it is minor.

We want go give power users an official short alias, irrespective of tab-completion, just like we do with `-Args` for `-ArgumentList`.

> add docs and comments too.

Docs, blog posts, and comments are only part of the puzzle: the parameter names themselves shouldn't be counterintuitive:

We can't fix the `-ArgumentList` / `-Args` name anymore (except to de-emphasize `-ArgumentList` in favor of `-ArgumentString`), but we can be more descriptive in the _new_ parameter's name:

`-Arguments` is ambiguous, `-IndividualArguments` is not.

**Comment by iSazonov** [comment]
With tab completion `Arguments` is more discoverable and more easy for adoption. Everything else does not make sense until the user reads the parameter description.

**Comment by mklement0** [comment]
If your concern is that `-IndividualArguments` doesn't start with the word "Argument", here's another option:

`-ArgumentArray` / `-Arga` 

Note that the `s` in the existing `-Args` can then be interpreted as referring to the "String" part of the renamed `-ArgumentString`

Here's a quick proof of concept:

```powershell
Add-Type @'
    using System;
    using System.Management.Automation;

    [Cmdlet("Start", "Process", DefaultParameterSetName = "ArgString")]
    public class StartProcessCommand : PSCmdlet {

        [Parameter(Position = 0)]
        public string FilePath { get; set; }

        [Parameter(ParameterSetName = "ArgArray", Position = 1)]
        [Alias("Arga")]
        public string[] ArgumentArray { get; set; }

        [Parameter(ParameterSetName = "ArgString", Position = 1)]
        [Alias("Args", "ArgumentList")]
        public string ArgumentString { get; set; }

        protected override void ProcessRecord() {
          WriteObject(ParameterSetName);
        }
    }
'@ -PassThru | % Assembly | Import-Module
```

Typing `Start-Process foo -a<tab>` then cycles through the parameters in the following order:
`-ArgumentArray`, `-Arga`, `-ArgumentString`, `-Args`,  and  `-ArgumentList`

**Comment by SteveL-MSFT** [comment]
@PowerShell/powershell-committee need to review a new proposed parameter to accept an array of args passed as an array of args

**Comment by SteveL-MSFT** [comment]
@PowerShell/powershell-committee reviewed this, we would prefer to have a switch rather than a new parameter that changes the behavior to `-ArgumentList` rather than cause confusion to users between `-ArgumentList` vs `-ArgumentArray`.  However, we couldn't come up with a good switch name to describe the behavior to the user without being overly verbose so open to any suggestions.

**Comment by iSazonov** [comment]
New behavior is a preferred behavior so the new switch looks like an extra parameter in scripts. 
How are we going to get rid of it after the transition period? Maybe revert the switch logic? For backward compatibility users could add the switch.
Although I would just prefer the new parameter ArgumentArray. I don't think that it will confuse users if we explicitly say in helps that user should use ArgumentArray instead of ArgumentList.

**Comment by rjmholt** [comment]
The path through a new parameter rather than a switch seems clearer to me:

- Add a new parameter `ArgumentArray` (or `Arguments`, or whatever) with a separate parameter set
- Positional binding continues to be the old parameter
- In a later PowerShell version we make the breaking change to make the positional argument refer to the new parameter

**Comment by mklement0** [comment]
I also think that a new switch is the wrong way to go.

Note that the feared name confusion could be minimized based on the [above proposal](https://github.com/PowerShell/PowerShell/issues/5576#issuecomment-666566501): rename `-ArgumentList` to `-ArgumentString` and make `-ArgumentList` an _alias_ for it, for _programmatic_ backward compatibility.

With the renamed parameter, the syntax diagram would then only show `-ArgumentString`, which can be documented as such.

**Comment by Artoria2e5** [comment]
The new parameter thing sounda good to me. Adding it to my PR to reduce the scope of that experimental switch.

**Comment by trivalik** [comment]
I do not think that many scripts will break, since if it does not work, nobody would do it. If you where fixing it 3 years ago, now nobody would care about it, since Window 10 probably would by default use this version.

**Comment by mklement0** [comment]
> The only issue I have is that powershell Start not return values.

To get direct results, do _not_ use `Start-Process` - use _direct invocation_ instead:
`regsvr32.exe -ArgumentList 'C:\Some Path With Spaces\file.ocx' /s` or, in the context of a CLI call: 
`powershell -c  "regsvr32.exe 'C:\Some Path With Spaces\file.ocx' /s"`- see https://github.com/MicrosoftDocs/PowerShell-Docs/issues/6239

Your command is an example of a _workaround_, which _shouldn't be necessary_: Leaving the extra layer of `"` escaping required due to the use of PowerShell's _CLI_ out of the picture, your command amounts to:

```powershell
Start-Process regsvr32.exe -ArgumentList '"C:\Some Path With Spaces\file.ocx"', '/s'
```

This _embedded_ double-quoting shouldn't be necessary, because `Start-Process` should perform all necessary quoting automatically, behind the scenes, when it assembles the _single-string_ argument list for the target process (of necessity, on Windows). That is, you should be able to simply call:

```powershell
Start-Process regsvr32.exe -ArgumentList 'C:\Some Path With Spaces\file.ocx', '/s'
```
That is, all you should have to focus on is to satisfy PowerShell's _own_ syntax requirements.

If we changed the behavior of `-ArgumentList` this way, existing workarounds such as yours would break.
Hence the need for a _new_ parameter (and _not_ fixing the current, broken behavior doesn't seem like a satisfactory solution to me).

**Comment by trivalik** [comment]
There are ways:
- use a environment variable to opt out the new behavior
- add workaround by checking first or in whole string that 2 characters are \"  then use old behavior

Anyway I am for a break. If this break where already 3 years in past, probably nobody would complain now.

Thanks for you tip, but I need to elevate the process.

**Comment by ringerc** [comment]
The underlying issue here has been improved somewhat in Powershell 7.2 by the experimental `PSNativeCommandArgumentPassing` feature, which makes the `&` command preserve quoting.

Unfortunately it does *not* affect `Start-Process`, which retains its `-ArgumentList` string-splatting.

It's a real shame that `Start-Process` was not updated along with `&` to respect `$PSNativeCommandArgumentPassing`. Example:

```
#!pwsh
#Requires -Version 7.0.0
Set-StrictMode -version 3
$ErrorActionPreference = "Stop"
$VerbosePreference = "Continue"

# Make sure sane command passing is available, force shell restart if not
if (-Not (Get-ExperimentalFeature -Name PSNativeCommandArgumentPassing).Enabled) {
        Write-Error "run `"Enable-ExperimentalFeature -Name PSNativeCommandArgumentPassing`" and retry"
        exit 1
}

# Make "&" preserve quoting (but not unfortunately StartProcess -ArgumentList)
$PSNativeCommandArgumentPassing = "Standard"

# Works but we have to enable an experimental feature first, and it doesn't
# seem to be possible to enable it within the same shell that uses it...
Write-Output "& with PSNativeCommandArgumentPassing=Standard"
&"printf" "%s\n" "arg1" "arg2 with spaces" "arg3 `"with quotes`" in it" "arg4 `${metachars:?shellexec} embedded"

# Still doesn't work. Sigh.
Write-Output "`nStart-Process -ArgumentList"
Start-Process -FilePath "printf" -ArgumentList @("%s\n", "arg1", "arg2 with spaces", "arg3 `"with quotes`" in it", "arg4 `${metachars:?shellexec} embedded") -Wait -Passthru
```

**Comment by jredfox** [comment]
confirmed I tried 20 variations of this
```
start-process powershell -ArgumentList "`"C:\Users\jredfox\Desktop\script s.ps1`""
```

**Comment by jredfox** [comment]
also calling it straight from java with powershell.exe /c needing to specify which exe to process the command the rest works in the powershell UI but not from java. if you were able to get the same characters from cmd you could also reproduce. powershell is broken for the start-process and maybe in general
```java
		ProcessBuilder pb = new ProcessBuilder(new String[]
				{
						"powershell",
						"/c",
						"start-process",
						"powershell",
						"-ArgumentList",
						"'-File',",
						"'\"C:/Users/jredfox/Desktop/spacing test.ps1\"',",
						"'-ExecutionPolicy',",
						"'Bypass'",
						"-NoNewWindow"
				}).inheritIO();
		Process p = pb.start();
		while(p.isAlive());
		for(String s : pb.command())
			System.out.print(s + " ");
			```

**Comment by ringerc** [comment]
@jredfox The backstory here is that Windows itself has absolutely insane command-line handling. See this classic 2011 blog, which still describes the state of the art for the Win32 API: https://docs.microsoft.com/en-us/archive/blogs/twistylittlepassagesallalike/everyone-quotes-command-line-arguments-the-wrong-way 

This makes it very hard for Powershell to do the right thing in a general purpose way, so they just punted on the problem entirely, like the underlying .NET process creation calls did.

The pwsh 7.2 experimental feature I linked to above improves this for "bare" command invocation and for the "&" operator, but has no effect on `Start-Process`, so you cannot benefit from it in places where you want the control offered by `Start-Process`.

**Comment by igorakkerman** [comment]
Late to the party but the presented workarounds didn't work for me, so here's mine.

I wanted to execute a ps1 with spaces in its name as an administrator. What ultimately worked was to
1. write the argument list as a single string with all parameters
1. wrap the script name with double quotes
1. wrap the parameter string with single quotes *and* parantheses

```powershell
Start-Process -Verb runAs pwsh -ArgumentList ('-NoExit "path/to/my script with spaces.ps1"')
```

Needless to say it is very unsatisfying that after many years and major versions this issue has not been solved.

**Comment by ringerc** [comment]
@igorakkerman I agree. I understand that Powershell cannot handle every insane command-line argument parsing implementation that Windows programs may concoct, but there are only three argument parsing models that _really_ matter:

* MSVC runtime `CommandLineToArgV` parsing or compatible implementations from other C runtimes - this will cover 99% of cases on Windows since almost every C, C++, or .NET (C# etc) program uses this; and
* `cmd.exe`'s wacky command parsing with `%--` and similar magic; this matters since one shell invoking another is going to be important for powershell users; and
* Every non-windows OS, in which Powershell can just `execv()` the process for perfectly reliable, straight-forward argument vector processing that works perfectly for every kind of process

So long as Powershell retains a way to compose a `CreateProcessEx` raw argument string for weird corner cases like commands with their own custom argument parsers, it can meet all requirements. If it had some kind of `-ArgumentQuotingStyle PlatformDefault` option that'd cover 99% of uses with absolutely no fuss.

Handwaving it away with the argument that argument quoting and escaping is too hard is ... unfortunate. Python's `shlex` and `subprocess` modules handle it just fine. It's inconvenient and annoying but not fundamentally a hard problem, and the fact that PowerShell leaves this up to its individual users to work around made me give up on it completely for my automation needs.

**Comment by SteveL-MSFT** [comment]
@PowerShell/wg-powershell-cmdlets discussed this issue and have decided to take a look at this problem more generally on how to have `vNext` versions of cmdlets that addresses known issues without breaking existing usage or continuing to add unnecessary complexity to the cmdlet syntax to bridge both worlds.  We discussed the PSResourceGet case where it was a renamed module with renamed cmdlets so that it works side-by-side and considering if that is the pattern we should apply generally as a way to get to a more ideal case.

**Comment by SteveL-MSFT** [comment]
I created https://github.com/orgs/PowerShell/discussions/21089 to discuss this broader topic

**Comment by mklement0** [comment]
Note that the link in the previous comment no longer works; here's the current one:

* https://github.com/PowerShell/PowerShell/discussions/21089

**Comment by viocha** [comment]
It's hard to imagine that this problem still hasn't been solved after all these years. This is the solution provided by ChatGPT.
```pwsh
$argList = $argList | ForEach-Object { $_ | ConvertTo-Json }
```

**Comment by cbryn** [comment]
This is still open.

**Comment by noamper** [comment]
Still open

**Comment by ringerc** [comment]
There's still no correct, portable way to start a process in Powershell, so this issue remains open.
